code/modules/paperwork/ticketmachine.dm

```diff
@@ -0,0 +1,201 @@
+//Bureaucracy machine!
+//Simply set this up in the hopline and you can serve people based on ticket numbers
+
+/obj/machinery/ticket_machine
+	name = "ticket machine"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "ticketmachine"
+	desc = "A marvel of bureaucratic engineering encased in an efficient plastic shell. It can be refilled with a hand labeler refill roll and linked to buttons with a multitool."
+	density = FALSE
+	maptext_height = 26
+	maptext_width = 32
+	maptext_x = 7
+	maptext_y = 10
+	layer = HIGH_OBJ_LAYER
+	var/ticket_number = 0 //Increment the ticket number whenever the HOP presses his button
+	var/current_number = 0 //What ticket number are we currently serving?
+	var/max_number = 100 //At this point, you need to refill it.
+	var/cooldown = 50
+	var/ready = TRUE
+	var/list/ticket_holders = list()
```

Comment:
This needs to be `Cut` in a `Destroy` proc.

Reply:
Fixed